### DISCUSSION TITLE
Pluggable architecture for conditions in sitecore-jss-forms and sitecore-jss-react-forms packages

### DISCUSSION THREAD

**Discussion by Antonytm** [Sitecore/jss#1404] [Ideas]
## Is your feature request related to a problem? Please describe.
**Support of conditions for sitecore-jss-react-forms.**

We want to have a form with a radio button at the end of the form. Depending on the selected value in this field, we want either show `submit form` button or `next page` button. You are able to add conditions to multipage Sitecore Forms using the Sitecore Forms wizard. You get these conditions in the Layout Service JSON. But they are ignored. 

Following documentation, 
> This module is intended to be copied from and extended as needed - it is not meant to be a fully-fledged implementation for rendering Sitecore Forms in JSS applications.

(It means that probably it was never planned that this part should work)

## Describe the solution you'd like
I understand that it could be a big task to implement all conditions that are possible to configure on the Sitecore Forms interface. That is why it would be nice to have a pluggable architecture for conditions. It will allow the implementation of conditions that you need by yourself and further upgrade of `sitecore-jss-forms` and `sitecore-jss-react-forms` NPM packages with no issues. Otherwise, during each upgrade, we will need to look at what was changed inside `sitecore-jss-forms` and `sitecore-jss-react-forms` and match this functionality on our side(or ignore it, but we can miss something important in this case)

## Describe alternatives you've considered
1. Don't use Sitecore Forms (use Google Forms or some other 3rd party provider)
2. Implement required conditions by ourselves. At least part of the features.

**Comment by illiakovalenko** [comment]
Hello @Antonytm
Thanks for the provided request, I've added this to our backlog for discussion

**Comment by shayanhussaini** [comment]
Is this feature(conditions in sitecore forms) available in the latest sitecore-jss package and work with next js app?
![Forms](https://github.com/user-attachments/assets/cb7bacb8-a04b-4aba-918c-3de00d618705)

**Reply by art-alexeyenko** [reply]
@shayanhussaini This feature is not implemented for Sitecore SXP forms yet. 
However, new Forms for XMCloud have been introduced for the XMC offering. These froms are handled by a separate library (not JSS) and every from condition set up on XMCloud side will be correctly displayed in JSS app.
